cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -174,7 +175,7 @@ public PushAppResponse pushApp(ApplicationReference appRef, String commitMessage
     // TODO CDAP-20371 revisit and put correct Author and Committer, for now they are the same
     CommitMeta commitMeta = new CommitMeta(committer, committer, System.currentTimeMillis(), commitMessage);
 
-    PushAppResponse pushResponse = sourceControlOperationRunner.push(new PushAppContext(appRef.getParent(), repoConfig,
+    PushAppResponse pushResponse = sourceControlOperationRunner.push(new PushAppRequest(appRef.getParent(), repoConfig,
```

Comment:
Let's name this one differently, `PushAppRequest` is already used in the `SourceControlManagementHttpHandler`

Reply:
what about PushAppOperation/PullAppOperation?

Reply:
Yeah, or `PushAppTaskRequest`/`PullAppTaskRequest`? As long as we can differentiate them

Reply:
I was thinking PushAppTaskRequest but it would be task specific. 
PushAppOperation feels to be a verb/function

How about PushAppOperationRequest?

Reply:
@samdgupi Yeah this one sounds good enough